rasa/core/actions/action.py

```diff
@@ -1098,8 +1098,7 @@ async def _run_custom_action(
             )
             for event in custom_events:
                 if isinstance(event, SlotSet):
-                    if tracker.get_slot(event.key) != event.value:
```

Comment:
This check exists as an optimization for the tracker, to prevent the tracker getting excessively large with duplicate `SlotSet` events. I'll familiarise myself with the ticket scenario and give my thoughts in a review early on Monday.

Reply:
Cool, that would work

Reply:
I've reviewed the ticket description, and this particular case of needing duplicate SlotSet events (either set by custom actions or entity extraction) in the case AugMemo Policy of truncating the trackers was not flagged when the default action `action_extract_slots` was being implemented. Considering this is causing this policy to not function properly, I'd say your PR change is an appropriate bugfix. 

If we want to make an improvement keeping the original optimisation goal of the tracker, we could introduce a new key in the domain yaml format to make this explicit (e.g. `keep_duplicate_slot_set_events: true/false`) 🤔 This would be outside of the scope of this ticket so I'd advise creating an improvement ticket to be evaluated and prioritised accordingly.

Reply:
@ancalita There's another question here that whether a SlotSet event be emitted when the Slot is set to `None`? This particular test `test_action_extract_slots_with_none_value_custom_mapping` seems to be failing for this reason -- it was expecting no events but with the changes in the PR an event was emitted

Reply:
I see 🤔  it's possible that the custom action resets slot(s) to `None` so there could be situations when the AugMemo policy requires this and we should have this `SlotSet` event recorded for that user turn.